webrender_traits/src/display_item.rs

```diff
@@ -587,6 +587,12 @@ impl ClipId {
     }
 
     pub fn new(id: u64, pipeline_id: PipelineId) -> ClipId {
+        // We do this because it is very easy to create accidentally create something that
+        // seems like a root scroll node, but isn't one.
+        if id == 0 {
```

Comment:
Did you consider panicing instead? It seems to me that the user needs to call `root_scroll_node` in the first place